src/m_menu.c

```diff
@@ -3962,19 +3962,24 @@ static void M_CoerceFPSLimit(void)
     fpslimit = 0;
 }
 
+static void M_ResetScreen(void)
+{
+  reset_screen = true;
```

Comment:
The problem was simple - it was possible to call `V_Init` several times before `I_FinishUpdate`. 
However, I suggest:
- refactoring `I_InitGraphicsMode` for "initialize" and "reset" functions
- move most of the gamepad/mouse code from `i_video.c`
- remove `V_Init` and `screens[]` for Choco system like `I_VideoBuffer`

But it's probably better to do it after the release, right?

Reply:
Yes, this needs to be redone. It is still mostly taken from WinMBF and thus tailored around SDL1.x with some additions from Chocolate Doom which doesn't even allow to change rendering parameters im-game.

But you are right. If there is a bug, we fix it now. But everything else can wait after the release, which I would really like to get out of the door in short term.

Reply:
> * remove `V_Init` and `screens[]` for Choco like `I_VideoBuffer`

Rejected this idea?

Reply:
No, still on the to-do list. Got distracted by macOS stuff. Also a bit busy at the moment, maybe in a day or two.